model/task/task_test.go

```diff
@@ -1253,6 +1253,63 @@ func TestBulkInsert(t *testing.T) {
 	}
 }
 
+func TestGetTaskIdBetweenIds(t *testing.T) {
+	assert := assert.New(t)
+	assert.NoError(db.ClearCollections(Collection))
+	displayName := "cool-task-9000"
+	buildVarient := "bv"
+	requester := "r"
+	project := "proj"
+	tasks := []Task{}
+	for i := 1; i <= 20; i++ {
+		task := Task{
+			Id:                  "t" + fmt.Sprint(i),
+			DisplayName:         displayName,
+			BuildVariant:        buildVarient,
+			Requester:           requester,
+			Project:             project,
+			RevisionOrderNumber: i,
+		}
+		assert.NoError(task.Insert())
+		tasks = append(tasks, task)
+	}
+	t10, err := GetMidwayTask(tasks[0], tasks[19])
+	assert.NoError(err)
+	assert.NotNil(t10)
```

Comment:
This has to be `require` - the difference between `assert` and `require` is that `require` will stop if the condition is not met, whereas `assert` will continue. Since this uses `assert`, it'll continue even if `t10` is nil (and report the error at the end of the test), but then the line after this will panic.

Same with the others in this file.